File: c/diagonal_negativos.c
#include <stdio.h>

int main(){
    int n, qtdnegativos;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);

    int matriz[n][n];

    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
			printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &matriz[i][j]);
        }
    }

    printf("DIAGONAL PRINCIPAL:\n");

    for (int i=0; i<n; i++) {
        printf("%d ", matriz[i][i]);
    }

    qtdnegativos = 0;
    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            if (matriz[i][j] < 0) {
                qtdnegativos++;
            }
        }
    }

	printf("\nQUANTIDADE DE NEGATIVOS = %d\n", qtdnegativos);

    return 0;
}

File: python/diagonal_negativos.py
n: int; qtdnegativos: int

n = int(input("Qual a ordem da matriz? "))

matriz: [[int]] = [[0 for x in range(n)] for x in range(n)]

for i in range(n):
	for j in range(n):
		matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("DIAGONAL PRINCIPAL:")

for i in range(n):
	print(f"{matriz[i][i]} ", end="")

qtdnegativos = 0
for i in range(n):
	for j in range(n):
		if matriz[i][j] < 0:
			qtdnegativos = qtdnegativos + 1

print(f"\nQUANTIDADE DE NEGATIVOS = {qtdnegativos}")